Add CI/CD for Linux builds

j4lc opened the pull request:
This PR adds CI/CD for the Linux builds.

Exports:
- `msdfgen-linux-arm` for the Linux ARM build
- `msdfgen-linux` for the Linux x64 build
- `msdfgen-windows` for the Windows x64 build

---

Proofs: https://github.com/jae1911/msdfgen-ydms/actions/runs/14518541853

---

Windows build was made possible thanks to @AwesomeTornado.

Commit:
feat: add Linux builds

Really easy builds, just needs the libheif dependencies.

AwesomeTornado committed:
VS2022 auto upgrade project

AwesomeTornado commented:
I was able to get the windows build working with Microsoft VS 2022, after doing the recommended auto upgrade. All that changed was the Msdfgen.vcxproj file, where the target windows platform was changed from win 8.1 to win 10, and the platform toolset, which was changed from v140 to v143.

Would you like me to create a pull request which adds this? I am unsure if you also found this, but ran into more problems, or if what I did would fix your issues.

You can find my fork here by the way.
https://github.com/AwesomeTornado/msdfgen

j4lc commented:
I could rebase your changes into this PR and add you as a co-author, how does that sounds?

AwesomeTornado commented:
Sure, sounds great!

Commit:
Merge branch 'jae/feat/arm-builds' of github.com:jae1911/msdfgen-ydms into jae/feat/arm-builds

Commit:
feat: add Windows builds

Made possible by AwesomeTornado, thanks again.
See a44e483ce6e6c9c773953dcc7e4716d99594dba7

Commit:
fix: reconvert arguments from the ADO build file

I completely skipped it at first, this should work a bit better now.
Though not sure if runners have the proper libraries for this,
especially on Windows.

j4lc commented:
Completely forgot to do this earlier, but here it is, it builds on Windows as well now.

Thanks a lot @AwesomeTornado!
